java/code/src/com/redhat/rhn/frontend/xmlrpc/errata/ErrataHandler.java

```diff
@@ -1436,15 +1436,10 @@ public Errata publishAsOriginal(User loggedInUser, String advisory,
      */
     private Errata publish(Errata errata, List<Channel> channels, User user,
             boolean inheritPackages) {
-        Errata published = ErrataFactory.publish(errata);
         for (Channel chan : channels) {
-            List<Errata> list = new ArrayList<Errata>();
-            list.add(published);
-            published = ErrataFactory.publishToChannel(list, chan, user,
-                    inheritPackages).get(0);
-
+            errata = ErrataFactory.publishToChannel(List.of(errata), chan, user, inheritPackages).get(0);
```

Comment:
I know it's out of scope, but this is a super strange code. Create a list with one element, and then from the method returned list get the first element and call it again...

Reply:
I agree, but would leave that for future PRs.